#!/usr/bin/env python
PACKAGE = "racer"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("position_weight", double_t, 0, "Distance to trajectory weight", 0, 0, 1000)
gen.add("heading_weight", double_t, 0, "Heading angle alignment weight", 0, 0, 1000)
gen.add("velocity_weight", double_t, 0, "Velocity alignment weight", 0, 0, 1000)
gen.add("distance_to_obstacle_weight", double_t, 0, "Distance from obstacles weight", 0, 0, 1000)
gen.add("max_position_error", double_t, 0, "Max expected distance from trajectory", 0, 0, 100)
gen.add("velocity_undershoot_overshoot_ratio", double_t, 0, "Ratio between overshooting and undershooting reference velocity", 1.0, 0, 2.0)

exit(gen.generate(PACKAGE, "racer", "DWA"))